src/pages/app/posting_project_user/ProfilePage.js

```diff
@@ -121,40 +151,61 @@ const ProfilePage = (props) => {
   const dispatch = useDispatch();
-  //Load profileData from global store into 'selectedData' variable using useSelector() hook
-  const selectedData = useSelector(
+
+  //Load data from global store into variables variable using useSelector() hook
+  const selectedProject = useSelector(
+    (state) => state.profileReducer.projectData.props
+  );
+  const selectedProfile = useSelector(
     (state) => state.profileReducer.profileData.props
   );
+
   //Update the data from the 'selectedData' variable into the local variable
-  //If the data is not equal to undefined (this mean the data is fetched succesfully) then
-  //assign it into the local variable.
-  //else, execute the FetchData() function.
-  if (selectedData != undefined) {
-    profileData = selectedData;
-  } else FetchData();
-  console.log("This is ProfileData name", profileData.orgName); // For testing purpose
-  console.log("Profile data saved succesfully:", profileData); //For testing purpose
-  //Function to fetch data and dispatch it into the global store. I will move this is the profileAction.js later
+  //If the data is equal to undefined (not fetched succesfully) then execute the FetchData() function.
+  if (selectedProject != undefined) {
+    projectData = selectedProject;
+  } else FetchData("public-projects", `${testing_project_id}`);
+  console.log("Profile data saved succesfully:", profileData.orgName); //For testing purpose
+  //Similar for Profile Data
+  if (selectedProfile != undefined) {
+    profileData = selectedProfile;
+  } else FetchData("organization", `${testing_organization_id}`);
+  console.log("Profile data saved succesfully:", profileData.orgName); //For testing purpose
+
+  //Function to fetch data and dispatch it into the global store.
   //The function need to use async and await to be able to fetch.
-  async function FetchData() {
+  async function FetchData(collection, fetchId) {
     let fetchedData; //This will later be used to fetch data.
     console.log("Starting to fetch data: "); //For testing purpose
-    //try and catch (from the concept promises) is to catch error.
+    console.log("Params: ", collection, " + ", fetchId); //For testing purpose
+
     try {
       // Retrieve the data from Firestore Cloud database. Need to be 'await'
       await firebase_db
-        .collection("public-projects") //access public projects
-        .doc(`${testing_project_id}`) //access the docs with this id
+        .collection(collection) //access public projects
+        .doc(fetchId) //access the docs with this id
         .get() //not sure what this do
         .then((documentSnapshot) => (fetchedData = documentSnapshot.data())); //fetch data into a variable.
       console.log("Profile Data fetched succesfully", fetchedData); //For testing purpose
       const saveData = (props) => {
         //Dispatch this to the global store using "profileReducer"
         dispatch({
-          type: "CHANGE_TEXT",
+          type: collection,
```

Comment:
Because I cannot move my code outside my main functional component (the profilePage), I prefer not to code a separated function, it would make my code (which already looks terrible) fucking terrible. And I think since I only use this function twice in my code, it's okay to code like this. Also, actually I think it should be like this. You know, a function should be is reusable.